packages/profiling-node/src/integration.ts

```diff
@@ -24,8 +31,296 @@ function addToProfileQueue(profile: RawThreadCpuProfile): void {
+  private _chunkStartTraceID: string | undefined = undefined;
+
+  /**
+   * Called when the profiler is attached to the client (continuous mode is enabled). If of the profiler
+   * methods called before the profiler is initialized will result in a noop action with debug logs.
+   * @param client
+   */
+  public initialize(client: NodeClient): void {
+    this._client = client;
+  }
+
+  /**
+   * Recursively schedules chunk profiling to start and stop at a set interval.
+   * Once the user calls stop(), the current chunk will be stopped and flushed to Sentry and no new chunks will
+   * will be started. To restart continuous mode after calling stop(), the user must call start() again.
+   * @returns void
+   */
+  public start(): void {
+    if (!this._client) {
+      // The client is not attached to the profiler if the user has not enabled continuous profiling.
+      // In this case, calling start() and stop() is a noop action.The reason this exists is because
+      // it makes the types easier to work with and avoids users having to do null checks.
+      DEBUG_BUILD && logger.log('[Profiling] Profiler was never attached to the client.');
+      return;
+    }
+    if (this._chunkId || this._chunkTimer) {
+      DEBUG_BUILD &&
+        logger.log(
+          `[Profiling] Chunk with chunk_id ${this._chunkId} is still running, current chunk will be stopped a new chunk will be started.`,
+        );
+      this.stop();
+    }
+
+    const scope = getCurrentScope();
+    const isolationScope = getIsolationScope();
+
+    // Extract the trace_id from the current scope and assign start_timestamp as well as the new chunk_id.
+    this._chunkId = uuid4();
+    this._chunkStartTraceID = {
+      ...isolationScope.getPropagationContext(),
+      ...scope.getPropagationContext(),
+    }.traceId;
+    this._chunkStartTimestampMS = timestampInSeconds();
+
+    CpuProfilerBindings.startProfiling(this._chunkId);
+    DEBUG_BUILD && logger.log(`[Profiling] starting profiling chunk: ${this._chunkId}`);
+
+    this._chunkTimer = global.setTimeout(() => {
+      DEBUG_BUILD && logger.log(`[Profiling] Stopping profiling chunk: ${this._chunkId}`);
+      this.stop();
+      DEBUG_BUILD && logger.log('[Profiling] Starting new profiling chunk.');
+      setImmediate(this.start.bind(this));
+    }, this._chunkIntervalMS);
+
+    // Unref timeout so it doesn't keep the process alive.
+    this._chunkTimer.unref();
+  }
+
+  /**
+   * Stops the current chunk and flushes the profile to Sentry.
+   * @returns void
+   */
+  public stop(): void {
+    if (!this._client) {
+      DEBUG_BUILD &&
+        logger.log('[Profiling] Failed to collect profile, sentry client was never attached to the profiler.');
+      return;
+    }
+    if (!this._chunkId) {
+      DEBUG_BUILD &&
+        logger.log(`[Profiling] Failed to collect profile for: ${this._chunkId}, the chunk_id is missing.`);
+      return;
+    }
+    if (this._chunkTimer) {
```

Comment:
m: I think we should move this above the `this._client` and `this._chunkId` check to enforce that we always clean up the timer, even if the client gets in a weird state.